cpp/src/arrow/python/datetime.cc

```diff
@@ -262,6 +265,42 @@ int64_t PyDate_to_days(PyDateTime_Date* pydate) {
                             PyDateTime_GET_DAY(pydate));
 }
 
+// GIL must be held when calling this function.
+Status StringToTzinfo(const std::string& tz, PyObject** tzinfo) {
+  static const std::regex kFixedOffset("([+-])(0[0-9]|1[0-9]|2[0-3]):([0-5][0-9])$");
```

Comment:
This won't work for gcc 4.8.x (which we still support)

Reply:
Hmm, I thought I saw the regex header in other parts of the codebase.  But I went ahead and converted this to a function.